### PR TITLE
Trus 3099 - 5MLD questionx

### PR DESCRIPTION
**Pull request by JoshuaH01**

### CODE REVIEW

File: app/models/UserAnswers.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ import scala.util.{Failure, Success, Try}
 
 trait ReadableUserAnswers {
   val data: JsObject
-  val is5mldEnabled: Boolean = false
+  val is5mldEnabled: Boolean = true
```
Comment (jackheslop96) [inline app/models/UserAnswers.scala:27]: I would consider leaving this as `false` while 5mld is not enabled in production

File: app/pages/register/beneficiaries/other/mld5/CountryOfResidenceYesNoPage.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+final case class CountryOfResidenceYesNoPage(index: Int) extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ Beneficiaries \ OtherBeneficiaries \ index \ toString
+
+  override def toString: String = "countryOfResidenceYesNo"
+
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] =
+    value match {
+      case Some(true) => userAnswers.set(CountryOfResidencePage(index), "GB")
```
Comment (jackheslop96) [inline app/pages/register/beneficiaries/other/mld5/CountryOfResidenceYesNoPage.scala:34]: `"GB"` available from https://github.com/hmrc/register-trust-beneficiary-frontend/blob/871ade38801e6f57cc47f4cb91f9341312ec0ff9/app/utils/Constants.scala#L20

File: app/pages/register/beneficiaries/other/mld5/UKResidentYesNoPage.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+final case class UKResidentYesNoPage(index: Int) extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ Beneficiaries \ OtherBeneficiaries \ index \ toString
+
+  override def toString: String = "ukResidentYesNo"
+
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] = {
+    value match {
+      case Some(true) => userAnswers.remove(AddressInternationalPage(index))
+      case Some(false) => userAnswers.remove(AddressUKPage(index))
```
Comment (jackheslop96) [inline app/pages/register/beneficiaries/other/mld5/UKResidentYesNoPage.scala:36]: Is this correct? I would have thought answering `yes` would cleanup country of residency and `no` would require no cleanup

File: app/utils/answers/OtherBeneficiaryAnswersHelper.scala [hunk 4]
```diff
@@ -57,6 +59,9 @@ class OtherBeneficiaryAnswersHelper @Inject()(answerRowConverter: AnswerRowConve
 
     Seq(
       bound.stringQuestion(DescriptionPage(index), "otherBeneficiary.description", DescriptionController.onPageLoad(index, draftId).url),
+      bound.yesNoQuestion(UKResidentYesNoPage(index), "otherBeneficiary.ukResidentYesNo", controllers.register.beneficiaries.other.mld5.routes.UKResidentYesNoController.onPageLoad(index, draftId).url),
+      bound.yesNoQuestion(CountryOfResidenceYesNoPage(index), "otherBeneficiary.countryOfResidenceYesNo", controllers.register.beneficiaries.other.mld5.routes.CountryOfResidenceYesNoController.onPageLoad(index, draftId).url),
```
Comment (jackheslop96) [inline app/utils/answers/OtherBeneficiaryAnswersHelper.scala:63]: Wrong way round? Should CountryOfResidenceYesNoPage not come first?

File: conf/messages.cy [hunk 5]
```diff
@@ -34,11 +34,26 @@ charity.5mld.countryOfResidenceYesNo.hint = Y wlad breswyl fel arfer yw’r wlad
...
+
+otherBeneficiary.countryOfResidenceYesNo.title = Do you know the other beneficiary’s country of residence?
+otherBeneficiary.countryOfResidenceYesNo.heading = Do you know {0}’s country of residence?
+otherBeneficiary.countryOfResidenceYesNo.checkYourAnswersLabel = Do you know {0}’s country of residence?
+otherBeneficiary.countryOfResidenceYesNo.error.required = Select yes if you know the charity’s country of residence
+
+otherBeneficiary.countryOfResidence.title = What is the other beneficiary’s country of residence?
+otherBeneficiary.countryOfResidence.heading = What is {0}’s country of residence?
+otherBeneficiary.countryOfResidence.checkYourAnswersLabel = What is {0}’s country of residence?
+otherBeneficiary.countryOfResidence.error.required = Enter a country
```
Comment (jackheslop96) [inline conf/messages.cy:59]: English messages in Welsh file?

File: conf/messages.en [hunk 6]
```diff
@@ -771,3 +771,21 @@ removeEmploymentRelatedBeneficiary.title = Are you sure you want to remove the e
...
+
+otherBeneficiary.ukResidentYesNo.title = Is this beneficiary a UK resident?
+otherBeneficiary.ukResidentYesNo.heading = Is {0} a UK resident?
+otherBeneficiary.ukResidentYesNo.checkYourAnswersLabel = Is {0} a UK resident?
+otherBeneficiary.ukResidentYesNo.error.required = Select yes if this beneficiary is a UK resident
+
+otherBeneficiary.countryOfResidenceYesNo.title = Do you know the other beneficiary’s country of residence?
+otherBeneficiary.countryOfResidenceYesNo.heading = Do you know {0}’s country of residence?
+otherBeneficiary.countryOfResidenceYesNo.checkYourAnswersLabel = Do you know {0}’s country of residence?
+otherBeneficiary.countryOfResidenceYesNo.error.required = Select yes if you know the charity’s country of residence
```
Comment (jackheslop96) [inline conf/messages.en:785]: charity?

File: conf/otherBeneficiary.routes [hunk 7]
```diff
@@ -19,9 +19,17 @@ POST        /:draftId/:index/uk-address                 controllers.register.ben
...
 GET         /:draftId/:index/check-details              controllers.register.beneficiaries.other.CheckDetailsController.onPageLoad(index: Int, draftId: String)
 POST        /:draftId/:index/check-details              controllers.register.beneficiaries.other.CheckDetailsController.onSubmit(index: Int, draftId: String)
 
 GET        /:draftId/:index/remove                     controllers.register.beneficiaries.other.RemoveOtherBeneficiaryController.onPageLoad(index: Int, draftId: String)
 POST       /:draftId/:index/remove                     controllers.register.beneficiaries.other.RemoveOtherBeneficiaryController.onSubmit(index: Int, draftId: String)
+
+GET         /:draftId/:index/uk-resident               controllers.register.beneficiaries.other.mld5.UKResidentYesNoController.onPageLoad(index: Int, draftId: String)
+POST        /:draftId/:index/uk-resident               controllers.register.beneficiaries.other.mld5.UKResidentYesNoController.onSubmit(index: Int, draftId: String)
+
+GET         /:draftId/:index/has-country-of-residence            controllers.register.beneficiaries.other.mld5.CountryOfResidenceYesNoController.onPageLoad(index: Int, draftId: String)
```
Comment (jackheslop96) [inline conf/otherBeneficiary.routes:31]: Very picky but not overly keen on `has-country-of-residence`. If that's how it's been done for the other types of beneficiary then fair enough for consistency's sake, but I think `know-country-of-residence` or `country-of-residence-yes-no` would be better. I'll leave it up to you.

Comment (JoshuaH01) [reply conf/otherBeneficiary.routes:31]: Spoke to Jack about this, leaving at the moment just following the pattern the other routes set

File: test/pages/register/beneficiaries/other/mld5/UKResidentYesNoPageSpec.scala [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        val answers: UserAnswers =
+          initial.set(AddressUKPage(index), UKAddress(str, str, Some(str), Some(str), str)).success.value
+            .set(AddressInternationalPage(index), InternationalAddress(str, str, Some(str), str)).success.value
+
+        val result = answers.set(UKResidentYesNoPage(index), true).success.value
+
+        result.get(CountryOfResidencePage(index)) mustNot be(defined)
+        result.get(AddressUKPage(index)) must be(defined)
+    }
+  }
```
Comment (jackheslop96) [inline test/pages/register/beneficiaries/other/mld5/UKResidentYesNoPageSpec.scala:49]: This test isn't doing anything. You're asserting that CountryOfResidencePage is undefined but not setting it in the first place. This test should initially set CountryOfResidencePage, then set UKResidentYesNoPage to true, then check that CountryOfResidencePage is undefined

**Pull request merged**